pandas/core/groupby/ops.py

```diff
@@ -224,6 +225,12 @@ def _get_cython_vals(self, values: np.ndarray) -> np.ndarray:
                 # result may still include NaN, so we have to cast
                 values = ensure_float64(values)
 
+            elif how == "sum":
```

Comment:
Good to add a comment why we make an exception for sum (AFAIU to match DataFrame/Series.sum)

Reply:
Mostly because we want to avoid the overflow, we are still casting back to int8 for example if possible. This would be a change in behavior we should deprecate before changing

Reply:
Added the comment